### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by mustafaiman**
Adds LoggingService
Adds DefaultLogger
Adds NoLogger
Set `config.properties['hazelcast.logging']` to `off` => Turn off logs(activate NoLogger)
Set `config.properties['hazelcast.logging']` to `default` => Turn on default logging (console.log at INFO level)
Set `config.properties['hazelcast.logging']` to an implementation of `ILogger` to use your own logger. One can easily provide adapters for existing logging libraries. There is a test which adapts `winston` in `LoggingTest.js`.

### CODE REVIEW

**Comment by gAmUssA** [comment]
:+1:

File: test/LoggingTest.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                new (winston.transports.Console)()
+            ]
+        }),
+
+        levels: [
+            'error',
+            'warn',
+            'info',
+            'debug',
+            'silly'
```
Comment (bilalyasar) [inline test/LoggingTest.js]: are  those levels standardised levels?

File: src/LoggingService.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import HazelcastClient = require('./HazelcastClient');
+import DefaultLogger = require('./DefaultLogger');
+import NoLogger = require('./NoLogger');
+export enum LogLevel {
+    ERROR = 0,
+    WARN = 1,
+    INFO = 2,
+    DEBUG = 3,
+    TRACE = 4
+}
```
Comment (bilalyasar) [inline src/LoggingService.ts]: do we need `ALL` log level as log4j includes?

File: test/LoggingTest.js (hunk 1 above)
Comment (mustafaiman) [inline test/LoggingTest.js]: those are winston logging library's levels. purpose of this code piece is to map our levels to winston's. Our levels are 
error
warn
info
debug
trace

File: src/LoggingService.ts (hunk 2 above)
Comment (mustafaiman) [inline src/LoggingService.ts]: isn't it same thing as TRACE in our case? When you set to TRACE it should show all logs as there is no other log level bigger than TRACE.

Comment (bilalyasar) [inline src/LoggingService.ts]: yes, you are right, highest level shows all logs.

File: test/LoggingTest.js (hunk 1 above)
Comment (bilalyasar) [inline test/LoggingTest.js]: ok, thanks.

**Comment by bilalyasar** [comment]
:+1:

**Comment by gAmUssA** [comment]
Guys, do we have a consensus here? 
Can it be merged @asimarslan  @GhostInAMachine ?

**Comment by mustafaiman** [comment]
@gAmUssA : I think @GhostInAMachine will take a look at it.

**Comment by gAmUssA** [comment]
dear @GhostInAMachine, kindly, review and merge

**Comment by gAmUssA** [comment]
@mustafaiman could you rebase so @GhostInAMachine can review after

**Comment by mustafaiman** [comment]
rebased. waiting for @GhostInAMachine to review

**Comment by GhostInAMachine** [comment]
👍

**Pull request merged**